src/gui/PasswordEdit.cpp

```diff
@@ -119,3 +142,81 @@ void PasswordEdit::autocompletePassword(const QString& password)
         setText(password);
     }
 }
+
+bool PasswordEdit::event(QEvent* event)
+{
+    if (!m_basePasswordEdit) {
+        // check caps lock state only for primary password edits
+        checkCapslockState();
+    }
+    return QLineEdit::event(event);
+}
+
+#include <iostream>
+void PasswordEdit::checkCapslockState()
+{
+    bool newCapslockState = m_capslockState;
+    Q_UNUSED(m_capslockState)
+    const QString platform = QGuiApplication::platformName();
+
+#if defined(Q_OS_WIN)
+    newCapslockState = (GetKeyState(VK_CAPITAL) == 1);
+#elif defined(Q_OS_MACOS)
+    newCapslockState = ((CGEventSourceFlagsState(kCGEventSourceStateHIDSystemState) & kCGEventFlagMaskAlphaShift) != 0);
+#elif defined(Q_OS_UNIX)
```

Comment:
`Q_OS_LINUX`? This would not apply to BSD or similar, would it?

Reply:
BSD also runs X11.